code/object/collideshipship.cpp

```diff
@@ -473,6 +473,36 @@ static bool check_subsystem_landing_allowed(ship_info *heavy_sip, collision_info
 	return false;
 }
 
+/**
+ * Helper function that calculates the next time this ship should be allowed to collide with a client player on multiplayer.
+ */
+static int calculate_next_multiplayer_client_collision_time(float impulse_magnitude)
+{
+	// Uses a continuous cubic rational function that allows us to space out collisions based on collision strength.
+	// based on some quick tests from Asteroth about how strong impuse is on average.
+	constexpr float IMPULSE_RANGE_FACTOR = 0.000000000001031f; 	// The inverse of the range of regular impulse values, 9900, to the third power
+	constexpr int MIN_IMPULSE = 100;							// The lowest values of collisions that we see.  Ignore if lower than this.
```

Comment:
If this is going to be used only with other floats, make this a `float` to save a type conversion.

Reply:
Should be good now.